### PR TITLE
Update to Python 3.9.5 and fix local development issues

### LINKED ISSUE #1296: numpy installation fails during Docker build
I'm attempting to build Tock locally and am running into issues during the Docker build: 

```shell
$ docker-compose build app 2>&1

Building app
#1 [internal] load build definition from Dockerfile
#1 sha256:181d0c374f79b0222092a0a804727c3f9c9c528e6cd6e30efc8aec9efdf2692c
#1 transferring dockerfile: 37B done
#1 DONE 0.0s

[snip... lots of output]

#10 463.3 [pipenv.exceptions.InstallError]: ERROR: Command errored out with exit status 1: /usr/local/bin/python -u -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-hhzjzwue/numpy_5bebe2a4c31d40739625560959037dbd/setup.py'"'"'; __file__='"'"'/tmp/pip-install-hhzjzwue/numpy_5bebe2a4c31d40739625560959037dbd/setup.py'"'"';f = getattr(tokenize, '"'"'open'"'"', open)(__file__) if os.path.exists(__file__) else io.StringIO('"'"'from setuptools import setup; setup()'"'"');code = f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record /tmp/pip-record-v3h_lqj5/install-record.txt --single-version-externally-managed --compile --install-headers /usr/local/include/python3.10/numpy Check the logs for full command output.
#10 463.3 ERROR: Couldn't install package: numpy
#10 463.3  Package installation failed...
#10 463.3 /usr/local/lib/python3.10/subprocess.py:1066: ResourceWarning: subprocess 20484 is still running
#10 463.3   _warn("subprocess %s is still running" % self.pid,
#10 463.3 sys:1: ResourceWarning: unclosed file <_io.FileIO name=6 mode='wb' closefd=True>
#10 463.3 sys:1: ResourceWarning: unclosed file <_io.FileIO name=7 mode='rb' closefd=True>
#10 ERROR: executor failed running [/bin/sh -c pipenv install --system --dev]: exit code: 1
------
 > [6/6] RUN pipenv install --system --dev:
------
executor failed running [/bin/sh -c pipenv install --system --dev]: exit code: 1
Service 'app' failed to build : Build failed
```

Near as I can tell this came in with 4722a68abe5ed9ef7da5d0c9a5e6653d9da703f7--if I revert that commit the build succeeds.

### PR DESCRIPTION
**Pull request by matthinz**
## Description

This PR addresses issues with getting a Tock development environment running locally using Docker. The local Docker build was failing (#1296) due to transitive dependencies not supporting the Python version in the `Dockerfile`. The fix there was moving the `Dockerfile` back to 3.9.5 (the latest stable version supported in the Cloud Foundry buildpack used in production).

While investigating that, I noticed there were a few places in the codebase that need to be touched when updating the Python version to ensure consistency between dev, CI, and production. I made those updates and noted the steps in `docs/dependency-management.md`.

Finally, once that was sorted, there was _another_  dependency issue caused by a recent update to `psycopg2-binary`. To resolve that I pinned to a prior version.

At this point, the local Docker environment is building and running successfully.

Closes #1296

### CODE REVIEW

**Comment by matthinz** [comment]
@kfoley-18F @neilmb Please let me know if I'm missing anything here--I'm relatively new to Python specifically but have plenty of dependency hell experience from the Node world.

File: runtime.txt [hunk 1]
```diff
@@ -1 +1 @@
-python-3.8.8
+python-3.9.6
```
Comment (neilmb) [inline runtime.txt:1]: I'm not convinced that Cloudfoundry's Python buildpack has 3.9.6 yet. I see https://github.com/cloudfoundry/python-buildpack/releases that lists up through 3.9.5, and I'm not totally sure how to make sure that cloud.gov is running the latest 1.7.42 release of that Python buildpack.

Can you try deploying this to `tock-staging` using the process in https://github.com/18F/tock/blob/main/docs/deployment-process.md and see if it is able to find Python 3.9.6 and get up and running?

Comment (matthinz) [reply runtime.txt:1]: I will do that. It looks like there's also an issue with finding 3.9.6 on circleci, so I will look into that as well.

Comment (matthinz) [reply runtime.txt]: I was able to verify that cloud.gov's buildpack does not support 3.9.6 by pushing to my personal sandbox. Looks like 3.9.5 is the most recently supported version. I updated the title & description to reflect this.

Comment (kewlguy781) [reply runtime.txt]: 3.9.6 is now listed there. Do we want to stay with 3.9.5 or go ahead and do 3.9.6? @matthinz @neilmb

Comment (neilmb) [reply runtime.txt]: I don't care either way. Take this PR as-is and do another to upgrade to 3.9.6? First figure out a policy on when and how we make these changes?

Comment (matthinz) [reply runtime.txt]: RE: Policy, it looks like there are two sources of Python version updates:

1. cloud.gov buildpack changes
2. Snyk security updates

It seems like cloud.gov changes will likely require an engineer periodically reviewing the state of staging/prod and updating. Snyk seems like it's good at poking us when new versions come out. I'm happy as long as the local dev environment stays in sync. I tried to address this process in [the docs](https://github.com/18F/tock/pull/1299/files#diff-3454e22d3d03c40d515cf0b06e7436f912886f8ae838228b6522e5454f6e518aR37)

Comment (kfoley-18F) [reply runtime.txt]: I got email from cloud.gov that they've updated their buildpacks today, so there may be additional versions available. (3.9.6)

Comment (kfoley-18F) [reply runtime.txt]: https://github.com/cloudfoundry/python-buildpack/releases

looks like 3.9.6 is now included...

**Review by kfoley-18F — APPROVED** [review]
I'm inclined to take the update to 3.9.5 now and make another issue to track 3.9.6. We can merge both into staging and validate behavior before pushing a release to production.

**Pull request merged**